test/cpp/api/serialize.cpp

```diff
@@ -64,7 +64,7 @@ void is_optimizer_state_equal(
 }
 
 template <typename OptimizerClass, typename DerivedOptimizerOptions, typename DerivedOptimizerParamState>
-void test_serialize_optimizer(DerivedOptimizerOptions options) {
+void test_serialize_optimizer(DerivedOptimizerOptions options, int state_size = 2) {
```

Comment:
I think it would be better if we make it `bool only_has_global_state` instead, because the caller of this function shouldn't need to know about the expected state size, and it should only need to tell this function whether the optimizer only has global state

Reply:
good idea. will do